NotificationHubs/src/com/windowsazure/messaging/Notification.java

```diff
@@ -101,13 +101,29 @@ public static Notification createAppleNotification(String body, Date expiry) {
         return n;
     }
 
+	/**
+	 * Utility method to set up a native notification for APNs.
+	 * Allows setting the APNS Headers - as per the new APNs protocol.
+	 * @param body
+	 * @param headers - APNs headers
+	 * @return
+	 */
+	public static Notification createAppleNotification(String body, Map<String, String> headers) {
+		Notification n = new Notification();
+		n.body = body;
+		n.contentType = ContentType.APPLICATION_JSON;
+		n.headers = headers;
+		n.headers.put("ServiceBusNotification-Format", "apple");
+		return n;
+	}
+
 	/**
 	 * Utility method to set up a native notification for GCM.
 	 * 
 	 * @param body
 	 * @return
 	 */
-	public static Notification createGcmNotifiation(String body) {
+	public static Notification createGcmNotification(String body) {
```

Comment:
Instead of having the new method be called `Gcm`, lets say `Fcm` as Google has deprecated the old mechanism.